src/main/java/io/stargate/sgv2/jsonapi/service/reranking/configuration/RerankingProviderConfigProducer.java

```diff
@@ -71,6 +72,11 @@ RerankingProvidersConfig produce(
     }
 
     validateRerankingProvidersConfig(rerankingProvidersConfig);
+
+    // Initialize the default reranking provider and model in DefaultRerankingProviderDef as
+    // Singleton.
+    CollectionRerankDef.initializeDefaultRerankDef(rerankingProvidersConfig);
```

Comment:
I wish we did not need to do this -- relying on state of Quarkus initialization. But I understand why it is necessary for what we need.